### ISSUE TITLE
Template part transforms

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#29249]
I think it may be worth discussing and exploring the transform options for template parts. Currently the transform menu includes options to transform template parts in to Columns or Group blocks:

<img width="392" alt="Screenshot 2021-02-23 at 10 14 43" src="https://user-images.githubusercontent.com/846565/108829485-229fca00-75c0-11eb-9dd3-1be96c741242.png">

However, neither of this options work. Selecting one just kicks you back to the wp-admin Dashboard 🐛 

Should it be possible to transform template parts into other, non-template-part blocks at all? This seems like a potentially dangerous operation that might be better left to the "Detach blocks from template part" option in the ellipsis menu:

<img width="554" alt="Screenshot 2021-02-23 at 10 20 46" src="https://user-images.githubusercontent.com/846565/108830037-d0ab7400-75c0-11eb-8958-6c42456733b3.png">

---

One transformation that should be possible in one way or another is switching template part variations. In previous issues the following design has been posited as an option to do this:

<img src="https://user-images.githubusercontent.com/846565/106589428-eb4f7780-6543-11eb-9ec6-f8e81d790a69.png">

---

Closely related – in #28737 we are exploring how patterns that are contextually relevant to the selected template might be exposed via the transform menu.

---

Finally, in https://github.com/WordPress/gutenberg/pull/27397#issuecomment-783315207 @mtias questioned whether it should be possible to quickly/easily wrap a template part inside another block. As @jasmussen mentioned, this could be useful for things like Sidebar template parts. My personal feeling is that this could possibly be better handled by the aforementioned patterns flow, as it is not a transform in the traditional sense, but it is worth discussing.

### Tentative action plan

- [ ] Remove Group/Columns transform options on Template Part block – #29296
- [x] Add Template Part Switching
- [ ] Provide a way to view block patterns that are contextually relevant to the selected Template Part – #28737
- [ ] Potentially create an affordance for template parts to be wrapped in other blocks

**Comment by carolinan** [comment]
Image 3:
Is it currently possible to create _block variations_ of template parts in the editor/UI?

Or only by registering the block variations in the theme? Registering variations in a theme is more complex than creating a header in the editor and saving it.

Or are these separate template parts (as HTML files or created in the editor), not _block variations_ ?

**Comment by jameskoster** [comment]
Apologies, probably poor word choice there on my part. They are variations in spirit, but not technically _block variations_.

> Is it currently possible to create block variations of template parts in the editor/UI?

Once https://github.com/WordPress/gutenberg/pull/29159 lands, it will be possible to assign multiple template parts to a single `area`. These would consequently be switchable per template as described in the OP.

**Comment by overclokk** [comment]
> Should it be possible to transform template parts into other, non-template-part blocks at all?

No, This should not do any transform, a template parts should only include a template and nothing more, "Detach blocks from template part" option in the ellipsis menu and a switch option I think they are more interesting in this kind of block.

**Comment by jameskoster** [comment]
Closing this in favor of https://github.com/WordPress/gutenberg/issues/31750
